include/itkBoxSpatialObjectToVTKPolyDataFilter.hxx

```diff
@@ -0,0 +1,142 @@
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+#ifndef itkBoxSpatialObjectToVTKPolyDataFilter_hxx
+#define itkBoxSpatialObjectToVTKPolyDataFilter_hxx
+
+#include "itkBoxSpatialObjectToVTKPolyDataFilter.h"
+
+#include <vtkCubeSource.h>
+
+namespace itk
+{
+
+template< unsigned int VDimension >
+BoxSpatialObjectToVTKPolyDataFilter<VDimension>
+::BoxSpatialObjectToVTKPolyDataFilter()
+{
+  this->SetNumberOfRequiredInputs(1);
+  this->SetNumberOfRequiredOutputs(1);
+}
+
+
+template< unsigned int VDimension >
+BoxSpatialObjectToVTKPolyDataFilter<VDimension>
+::~BoxSpatialObjectToVTKPolyDataFilter()
+{
+}
+
+
+template< unsigned int VDimension >
+void
+BoxSpatialObjectToVTKPolyDataFilter<VDimension>
+::SetInput( const BoxSpatialObjectType * input )
+{
+  // Process object is not const-correct so the const_cast is required here
+  this->ProcessObject::SetNthInput( 0, const_cast< BoxSpatialObjectType * >( input ) );
+}
+
+
+template< unsigned int VDimension >
+const typename BoxSpatialObjectToVTKPolyDataFilter<VDimension>::BoxSpatialObjectType *
+BoxSpatialObjectToVTKPolyDataFilter<VDimension>
+::GetInput() const
+{
+  return itkDynamicCastInDebugMode< const BoxSpatialObjectType * >( this->GetPrimaryInput() );
+}
+
+
+template< unsigned int VDimension >
+const typename BoxSpatialObjectToVTKPolyDataFilter<VDimension>::BoxSpatialObjectType *
+BoxSpatialObjectToVTKPolyDataFilter<VDimension>
+::GetInput( unsigned int idx ) const
+{
+  return dynamic_cast< const BoxSpatialObjectType * >( this->ProcessObject::GetInput(idx) );
+}
+
+
+template< unsigned int VDimension >
+vtkPolyData *
+BoxSpatialObjectToVTKPolyDataFilter<VDimension>
+::GetOutput()
+{
+  return this->m_PolyData.GetPointer();
+}
+
+
+template< unsigned int VDimension >
+void
+BoxSpatialObjectToVTKPolyDataFilter<VDimension>
+::PrintSelf(std::ostream & os, Indent indent) const
+{
+  Superclass::Superclass::PrintSelf(os, indent);
+
+  os << indent << "PolyData output: ";
+    this->m_PolyData->Print( os );
+}
+
+
+template< unsigned int VDimension >
+void
+BoxSpatialObjectToVTKPolyDataFilter<VDimension>
+::Update()
+{
+  if( this->GetInput()->GetMTime() > this->m_ProcessedInputMTime )
+    {
+    BoxSpatialObjectType * input = const_cast< BoxSpatialObjectType * >( this->GetInput() );
+    input->Update();
+    this->GenerateData();
+    this->m_ProcessedInputMTime = input->GetMTime();
+    }
+}
+
+
+template< unsigned int VDimension >
+void
+BoxSpatialObjectToVTKPolyDataFilter<VDimension>
+::GenerateData()
+{
+  const BoxSpatialObjectType * spatialObject = this->GetInput();
+
+  this->m_PolyData = vtkSmartPointer< vtkPolyData >::New();
+
+  vtkNew< vtkCubeSource > cubeSource;
+  std::cout << "size: " << spatialObject->GetSize() << std::endl;
```

Comment:
The box exists as a box in the "object" coordinates in which it is defined.   This is prior to applying the "GetObjectToWorldTransform()" that defines how that box should appear in "World" coordinates.   World coordinates are the composition of the transforms in the SpatialObject parent-child hierarchy that leads to the object.   A box, for example, can exist within an atlas, that has been registered to a patient, who is on a surgical table.  The box keeps track of its size and position in its own "object" space.   If you want to create a vtk representation of the object, you probably want to use its coordinates in "World" space.   You will need to query for each corner of the box in object space (defined by its position and size) and then transform those corners into world coordinates using the GetObjectToWorldTransform().   Note that since we allow affine transforms, the box may not remain a box after the ObjectToWorldTransform.

Hope this helps.   Now is a great time to give suggestions for helper functions - perhaps BoxSpatialObject should have a member function to return corner coordinates in World space?

Reply:
One idea - we can have SpatialObjects implement "Update" functions in which they compute their representations in World coordinates and make those representations available via appropriate member functions.

For BoxSpatialObject, the Update function would compute the corners of the box in World coordinates, and those corners can be accesses via GetCorner( unsigned int cornerNumber) and GetCorner(void).

Reply:
For clarity, we can change  SetSize to SetObjectSize and SetPosition to SetObjectPosition and define GetWorldCorner(..) and GetWorldCorners(..).

Reply:
Phenomenal explanation -- extremely helpful, thanks!

Considering this, what do you think of applying `ObjectToWorldTransform` with [`vtkTransformPolyDataFilter`](https://vtk.org/doc/nightly/html/classvtkTransformPolyDataFilter.html)?

Reply:
That would definitely be valid, and I think it is the best solution!   For PointBasedSpatialObjects (such as Tubes, Surfaces, and Blobs), this is definitely the most scalable solution.   Thanks!